cache/lru_cache.cc

```diff
@@ -244,17 +253,13 @@ void* LRUCacheShard::operator new(size_t size) {
   return port::cacheline_aligned_alloc(size);
 }
 
-void* LRUCacheShard::operator new[](size_t size) {
-  return port::cacheline_aligned_alloc(size);
-}
+void* LRUCacheShard::operator new(size_t /*size*/, void* ptr) { return ptr; }
```

Comment:
The placement new is used now in LRUCache constructor. Since we override (normal) new, it will complaint the placement new variant is not defined. 

The normal new is also needed because we use new operator in tests and they need to be allocate through aligned_alloc(). Otherwise we need to replace each `new LRUCacheShard` with aligned_alloc.